### PR TITLE
fixed wrong selection offset in editor code

### PR DESCRIPTION
**Pull request by prudhvije**
Currently it's taking one line extra if selected complete line which results in inconsistency of suggestions.

### CODE REVIEW

**Comment by ScrapCodes** [comment]
Can you explain what was the bug, and how you have fixed it ?

**Comment by prudhvije** [comment]
In this code snippet if 1st line is completely selected then SelectionModel is returning 1 and 2 lines which gives wrong suggestions. This can be avoided by checking the SelectionModel end offset.

1.  Set<String> lines = new HashSet<String>();
2.  Document document = projectEditor.getDocument();

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/EditorDocOps.java [hunk 1]
```diff
@@ -45,10 +46,14 @@
...
+        SelectionModel selectionModel = projectEditor.getSelectionModel();
 
-        if (projectEditor.getSelectionModel().hasSelection()) {
-            head = document.getLineNumber(projectEditor.getSelectionModel().getSelectionStart());
-            tail = document.getLineNumber(projectEditor.getSelectionModel().getSelectionEnd());
+        if (selectionModel.hasSelection()) {
+            head = document.getLineNumber(selectionModel.getSelectionStart());
+            tail = document.getLineNumber(selectionModel.getSelectionEnd());
+            if (document.getLineStartOffset(tail) == selectionModel.getSelectionEnd()) {
+                tail--;
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/EditorDocOps.java]: Can you add comment why we need this here.

**Comment by ScrapCodes** [comment]
Can you update this PR soon ? it does not merge anymore.

**Pull request closed without merging**